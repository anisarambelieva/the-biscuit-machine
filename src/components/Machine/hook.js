import { useEffect, useState } from "react";

import * as constants from "../../constants.js";

const useMachine = () => {
  const [state, setState] = useState(constants.machineState.off);
  const [conveyorWorking, setConveyorWorking] = useState(false);
  const [motorWorking, setMotorWorking] = useState(false);
  const [stamping, setStamping] = useState(false);
  const [animationState, setAnimationState] = useState(
    constants.animationState.running
  );
  const [fasterHeating, setFasterHeating] = useState(false);
  const [bakedBiscuitsCount, setBakedBiscuitsCount] = useState(0);

  useEffect(() => {
    if (state === constants.machineState.paused)
      setAnimationState(constants.animationState.paused);
    else setAnimationState(constants.animationState.running);
  }, [animationState, state]);

  return {
    animationState,
    stamping,
    setStamping,
    state,
    setState,
    fasterHeating,
    setFasterHeating,
    conveyorWorking,
    setConveyorWorking,
    motorWorking,
    setMotorWorking,
    bakedBiscuitsCount,
    setBakedBiscuitsCount,
  };
};

export default useMachine;
