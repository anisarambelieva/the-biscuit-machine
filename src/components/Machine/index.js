import { useEffect, useState } from "react";
import { Container, Row, Col } from "react-bootstrap";

import * as constants from "../../constants.js";
import BiscuitLifecycle from "../Biscuit/BiscuitLifecycle";
import BiscuitContainer from "../Container";
import Extruder from "../Extruder";
import Stamper from "../Stamper";
import Switch from "../Switch";
import BoostSwitch from "../boostSwitch.js";
import Conveyor from "../conveyor.js";
import Motor from "../motor.js";
import Oven from "../oven.js";

const Machine = () => {
  const [state, setState] = useState(constants.machineState.off);
  const [conveyorWorking, setConveyorWorking] = useState(false);
  const [motorWorking, setMotorWorking] = useState(false);
  const [stamping, setStamping] = useState(false);
  const [animationState, setAnimationState] = useState(
    constants.animationState.running
  );
  const [fasterHeating, setFasterHeating] = useState(false);
  const [bakedBiscuitsCount, setBakedBiscuitsCount] = useState(0);

  useEffect(() => {
    if (state === constants.machineState.paused)
      setAnimationState(constants.animationState.paused);
    else setAnimationState(constants.animationState.running);
  }, [animationState, state]);

  return (
    <Container>
      <Row style={{ marginTop: "10px" }}>
        <Col md="1">
          <Extruder />
        </Col>

        <Col md="1" style={{ position: "relative" }}>
          <Stamper animationState={animationState} stamp={stamping} />
        </Col>

        <Col md={{ span: 2, offset: 1 }}>
          <Oven
            machineState={state}
            setMotorWorking={setMotorWorking}
            fasterHeating={fasterHeating}
          />
        </Col>
      </Row>

      <Row style={{ position: "relative" }}>
        {conveyorWorking && (
          <BiscuitLifecycle
            animationState={animationState}
            onStamp={setStamping}
            machineState={state}
            setBakedBiscuitsCount={setBakedBiscuitsCount}
            bakedBiscuitsCount={bakedBiscuitsCount}
          />
        )}
      </Row>

      <Row style={{ height: "75px" }}></Row>

      <Row style={{ marginTop: "10px" }}>
        <Col md="6">
          <Conveyor conveyorWorking={conveyorWorking} />
        </Col>
      </Row>

      <Row style={{ marginTop: "20px" }}>
        <Col md={{ span: 2, offset: 5 }}>
          <BiscuitContainer />
        </Col>
      </Row>

      <Row>
        <Col md="1">
          <Motor
            machineState={state}
            setMotorWorking={setMotorWorking}
            motorWorking={motorWorking}
            setConveyorWorking={setConveyorWorking}
          />
        </Col>

        <Col md={{ span: 2, offset: 2 }}>
          <Switch onChange={setState} />
        </Col>
      </Row>

      <Row>
        <BoostSwitch onChange={() => setFasterHeating(!fasterHeating)} />
      </Row>

      <Row>
        <div>{bakedBiscuitsCount} baked biscuits</div>
      </Row>
    </Container>
  );
};

export default Machine;
