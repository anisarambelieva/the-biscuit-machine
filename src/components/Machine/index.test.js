import { render } from "@testing-library/react";

import Machine from "./index.js";

describe("<Machine />", () => {
  it("should match snapshot", () => {
    const { container } = render(<Machine />);

    expect(container).toMatchSnapshot();
  });
});
