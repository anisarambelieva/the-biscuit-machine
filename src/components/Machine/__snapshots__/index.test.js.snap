// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Machine /> should match snapshot 1`] = `
<div>
  <div
    class="container"
  >
    <div
      class="row"
      style="margin-top: 10px;"
    >
      <div
        class="col-md-1"
      >
        <div
          class="extruder"
        >
          Extruder
        </div>
      </div>
      <div
        class="col-md-1"
        style="position: relative;"
      >
        <div
          class="stamper"
        >
          Stamper
        </div>
      </div>
      <div
        class="col-md-2 offset-md-1"
      >
        <div
          class="oven"
          style="border-bottom: 15px solid #f2f2f2;"
        >
          <p
            style="margin: 0px;"
          >
            0
          </p>
        </div>
      </div>
    </div>
    <div
      class="row"
      style="position: relative;"
    />
    <div
      class="row"
      style="height: 75px;"
    />
    <div
      class="row"
      style="margin-top: 10px;"
    >
      <div
        class="col-md-6"
      >
        <p
          class="conveyor"
        >
          Conveyor
        </p>
      </div>
    </div>
    <div
      class="row"
      style="margin-top: 20px;"
    >
      <div
        class="col-md-2 offset-md-5"
      >
        <p
          class="biscuitContainer"
        >
          Container
        </p>
      </div>
    </div>
    <div
      class="row"
    >
      <div
        class="col-md-1"
      >
        <p
          class="motor"
        >
          Motor: 
          turned off
        </p>
      </div>
      <div
        class="col-md-2 offset-md-2"
      >
        <div
          style="padding: 5px; background-color: rgb(250, 220, 217); border-radius: 5px;"
        >
          <p>
            Switch
          </p>
          <button
            class="sc-bdvvtL eWZskW btn btn-primary"
            style="margin-right: 5px; background-color: rgb(171, 130, 129); border: gray;"
            type="button"
          >
            On
          </button>
          <button
            class="sc-bdvvtL eWZskW btn btn-primary"
            style="margin-right: 5px; background-color: rgb(171, 130, 129); border: gray;"
            type="button"
          >
            Paused
          </button>
          <button
            class="sc-bdvvtL eWZskW btn btn-primary"
            style="background-color: rgb(171, 130, 129); border: gray;"
            type="button"
          >
            Off
          </button>
        </div>
      </div>
    </div>
    <div
      class="row"
    >
      <form
        class=""
      >
        <div
          class="form-check form-switch"
        >
          <input
            class="form-check-input"
            type="checkbox"
          />
          <label
            class="form-check-label"
            title=""
          >
            Faster heating
          </label>
        </div>
      </form>
    </div>
    <div
      class="row"
    >
      <div>
        0
         baked biscuits
      </div>
    </div>
  </div>
</div>
`;
